x-pack/metricbeat/module/panw/config.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package panw
+
+import (
+	"errors"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+)
+
+const (
+	ModuleName = "panw"
+)
+
+type Config struct {
+	HostIp    string `config:"host_ip"`
+	Port      uint   `config:"port"`
+	ApiKey    string `config:"apiKey"`
+	DebugMode string `config:"apiDebugMode"`
```

Comment:
I would remove `port` since `host` should be enough. You can create a small helper function to validate the format and get the port. Let's use snake case and you can add `required` tags for required config fields. 

```suggestion
	Host    string `config:"host"` validate:"required"`
	ApiKey    string `config:"api_key"` validate:"required"`
	DebugMode string `config:"api_debug_mode"`
```

Reply:
I wasn't sure about port: would network admin ever change the default https port on a firewall?